### ISSUE TITLE
cmd/compile: internal compiler error: cannot export CLOSURE (34) node

### ISSUE THREAD

**Issue by driusan** [golang/go#19543]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?

go tip under Travis CI 

```
$ go version
go version devel +7a9aa06 Tue Mar 14 02:34:38 2017 +0000 linux/amd64)
```
### What operating system and processor architecture are you using (`go env`)?

```
$ go env

GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/travis/gopath"
GORACE=""
GOROOT="/home/travis/.gimme/versions/go"
GOTOOLDIR="/home/travis/.gimme/versions/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build447196172=/tmp/go-build"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

### What did you do?

Ran go fmt, updated a godoc, and sent a PR to github.com/Harvey-OS/ninep. The test failed on tip with the error:

> protocol/types.go:41:0: internal compiler error: cannot export CLOSURE (34) node
> ==> please file an issue and assign to gri@

and the following stack trace:

```
goroutine 1 [running]:
runtime/debug.Stack(0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/runtime/debug/stack.go:24 +0x79
cmd/compile/internal/gc.Fatalf(0xad5c17, 0x47, 0xc421b17468, 0x2, 0x2)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/subr.go:175 +0x230
cmd/compile/internal/gc.(*exporter).expr(0xc421b178d8, 0xc42067c500)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1423 +0x392
cmd/compile/internal/gc.(*exporter).expr(0xc421b178d8, 0xc420a8f440)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1341 +0x8a1
cmd/compile/internal/gc.(*exporter).exprList(0xc421b178d8, 0xc42035efc0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1141 +0xd1
cmd/compile/internal/gc.(*exporter).expr(0xc421b178d8, 0xc4209eb050)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1367 +0x979
cmd/compile/internal/gc.(*exporter).stmtList(0xc421b178d8, 0xc42035efe0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1120 +0x14b
cmd/compile/internal/gc.(*exporter).stmt(0xc421b178d8, 0xc4209eaf30)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1505 +0x6c4
cmd/compile/internal/gc.(*exporter).stmtList(0xc421b178d8, 0xc42055d040)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:1118 +0xf5
cmd/compile/internal/gc.export(0xc4217c1000, 0x0, 0x0)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/bexport.go:391 +0x8d4
cmd/compile/internal/gc.dumpexport()
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/export.go:185 +0x8a
cmd/compile/internal/gc.dumpobj1(0x7ffea186208f, 0x3c, 0x3)
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/obj.go:93 +0xa56
cmd/compile/internal/gc.dumpobj()
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/obj.go:49 +0x61
cmd/compile/internal/gc.Main()
	/home/travis/.gimme/versions/go/src/cmd/compile/internal/gc/main.go:524 +0x1fd0
main.main()
	/home/travis/.gimme/versions/go/src/cmd/compile/main.go:50 +0x101
```

The failing build is at https://travis-ci.org/Harvey-OS/ninep/jobs/210962148. I'm not really sure what else would be useful, I'm just doing what I was told.

### What did you expect to see?

Rainbows and unicorns.

### What did you see instead?

Fire and brimstone.

**Comment by josharian** [comment]
Thanks! This is useful. Can you provide a link to the PR itself, or wherever is easiest for us to find the code for reproducing locally?

cc @griesemer

**Comment by mdlayher** [comment]
Assigning to @griesemer per:
> ==> please file an issue and assign to gri@

**Comment by josharian** [comment]
Also, if you see either rainbows or unicorns coming out of the compiler, please file a bug. Yikes.

**Comment by driusan** [comment]
The PR that's failing on tip is:

https://github.com/Harvey-OS/ninep/pull/22/

But it's not doing anything other than comments and formatting, so the problem is with the code that's already in Harvey-OS/ninep (which I'm not very familiar with, which is why I was trying to get a better grasp on it by adding docs..), not the change itself.

**Comment by josharian** [comment]
Appears to be fixed now, probably as a dup of #19705.
